src/libraries/System.Data.OleDb/src/DbConnectionOptions.cs

```diff
@@ -367,17 +367,29 @@ protected internal virtual string Expand()
                 if (!rootFolderEndsWith && !fileNameStartsWith)
                 {
                     // need to insert '\'
+#if NETCOREAPP
+                    fullPath = string.Concat(rootFolderPath + '\\', value.AsSpan(fileNamePosition));
+#else
                     fullPath = rootFolderPath + '\\' + value.Substring(fileNamePosition);
+#endif
                 }
                 else if (rootFolderEndsWith && fileNameStartsWith)
                 {
                     // need to strip one out
+#if NETCOREAPP
+                    fullPath = string.Concat(rootFolderPath, value.AsSpan(fileNamePosition + 1));
+#else
                     fullPath = rootFolderPath + value.Substring(fileNamePosition + 1);
+#endif
                 }
                 else
                 {
                     // simply concatenate the strings
+#if NETCOREAPP
+                    fullPath = string.Concat(rootFolderPath, value.AsSpan(fileNamePosition));
+#else
                     fullPath = rootFolderPath + value.Substring(fileNamePosition);
+#endif
```

Comment:
Similarly, I would suppress this analyzer in System.Data.OleDb.  This code is nowhere near a hot path.

Reply:
I don't have a strong opinion on that so let me revert my changes.